scripts/dts/extract_dts_includes.py

```diff
@@ -478,6 +478,24 @@ def main():
     create_aliases(root)
     create_chosen(root)
 
+    # Re-sort instance_id by reg addr
+    # Build a dict of dicts, first level is index by compat
+    # second level is index by reg addr
+    compat_reg_dict = defaultdict(dict)
+    for r in reduced:
```

Comment:
Only `reduced[r]` seems to be used, so could do

```python
for node in reduced.values():
    ...
```

Reply:
I need compat as well.

Reply:
Thinking like this:

```python
compat_reg_dict = defaultdict(dict)
for node in reduced.values():
    instance = node.get('instance_id')
    if instance and node['addr'] is not None:
        for compat in instance:
            reg = node['addr']
            compat_reg_dict[compat][reg] = node
```

Reply:
Done.